server/zanata-war/src/test/java/org/zanata/adapter/GettextAdapterTest.java

```diff
@@ -22,43 +22,131 @@
-                        Optional.absent());
-//        System.out.println(DTOUtil.toXML(resource));
+        Resource resource = parseTestFile("test-gettext.pot");
         assertThat(resource.getTextFlows()).hasSize(3);
-        assertThat(resource.getTextFlows().get(0).getContents()).isEqualTo(
-                ImmutableList.of("Line One"));
-        // TODO test IDs
+
+        assertThat(resource.getTextFlows().get(0).getContents()).containsExactly("Line One");
+        assertThat(resource.getTextFlows().get(1).getContents()).containsExactly("Line Two");
+        assertThat(resource.getTextFlows().get(2).getContents()).containsExactly("Line Three");
+    }
+
+    @Test
+    public void testGettextWithComment() {
+        String testComment1 = " translator-comments";
+        Resource resource = parseTestFile("test-gettext-comments.po");
+        assertThat(resource.getTextFlows()).hasSize(3);
+
+        PoHeader poHeader = (PoHeader) resource.getExtensions().iterator().next();
+        assertThat(poHeader.getComment()).isEqualTo(testComment1);
+
+    }
+
+    @Test
+    public void testGettextPlurals() {
+        Resource resource = parseTestFile("test-gettext-plurals.po");
+        assertThat(resource.getTextFlows()).hasSize(1);
+        assertThat(resource.getTextFlows().get(0).getContents()).containsExactly("%n file", "%n files");
+    }
+
+    @Test
+    public void testGettextFuzzyFlag() {
+        Resource resource = parseTestFile("test-gettext-flags.pot");
+        assertThat(resource.getTextFlows()).hasSize(2);
+
+        PotEntryHeader potEntryHeader = (PotEntryHeader) resource.getTextFlows().get(0).getExtensions().iterator().next();
+        assertThat(potEntryHeader.getFlags().get(0)).isEqualTo("fuzzy");
+
+    }
+
+
+    @Test
+    public void testGettextReference() {
+        Resource resource = parseTestFile("test-gettext-reference.pot");
+        assertThat(resource.getTextFlows()).hasSize(2);
+
+        PotEntryHeader potEntryHeader = (PotEntryHeader) resource.getTextFlows().get(0).getExtensions().iterator().next();
+        assertThat(potEntryHeader.getReferences().get(0)).isEqualTo("reference");
+    }
+
+    @Test
+    public void testTranslatedGettext() {
+
+        Resource resource = parseTestFile("test-gettext-untranslated.pot");
+
+        String firstSourceId = resource.getTextFlows().get(0).getId();
+        String secondSourceId = resource.getTextFlows().get(1).getId();
+        String thirdSourceId = resource.getTextFlows().get(2).getId();
+
+        TranslationsResource transResource = new TranslationsResource();
+        addTranslation(transResource, firstSourceId, "Carpeta padre", ContentState.Approved);
+        addTranslation(transResource, secondSourceId, "Asunto:", ContentState.Translated);
+        addTranslation(transResource, thirdSourceId, "Conectar", ContentState.NeedReview);
+
+        File originalFile = new File(filePath.concat("test-gettext-untranslated.pot"));
```

Comment:
I _think_ you can actually get rid of this and set the corresponding originalFile.toURI() to null - gettextadapter doesn't use it at all.